### PR TITLE
[l10n] Update Material shareButtonLabel

### PR DESCRIPTION
**Pull request by bleroux**
## Description

This PR updates the Material label for the share option, currently it is "**Share...**" (the expected label on iOS), on Android it should be "**Share**".

Native Android TextField screenshot:

Capture d’écran du 2023-11-20 17-19-02

## Related Issue

Step 2 for https://github.com/flutter/flutter/issues/138728

## Tests

Adds 1 test.

### CODE REVIEW

File: packages/flutter_localizations/lib/src/l10n/material_en.arb [hunk 1]
```diff
@@ -207,7 +207,7 @@
     "description": "The label for the Search Web button and menu items on iOS."
   },
 
-  "shareButtonLabel": "Share...",
+  "shareButtonLabel": "Share",
   "@shareButtonLabel": {
     "description": "The label for the Share button and menu items on iOS."
```
Comment (chunhtai) [inline packages/flutter_localizations/lib/src/l10n/material_en.arb:212]: The description is inaccourate. same for other buttonlabel in this file

File: packages/flutter_localizations/test/material/translations_test.dart [hunk 2]
```diff
@@ -535,9 +535,18 @@ void main() {
   // Regression test for https://github.com/flutter/flutter/issues/110451.
   testWidgets('Finnish translation for tab label', (WidgetTester tester) async {
     const Locale locale = Locale('fi');
-    expect(GlobalCupertinoLocalizations.delegate.isSupported(locale), isTrue);
+    expect(GlobalMaterialLocalizations.delegate.isSupported(locale), isTrue);
```
Comment (chunhtai) [inline packages/flutter_localizations/test/material/translations_test.dart:538]: why this change?

**Review by chunhtai — COMMENTED** [review]
This may need to go through google internal translation pipeline. @HansMuller Do you know who is taking care of that now? or is there a doc on the process?

File: packages/flutter_localizations/lib/src/l10n/material_en.arb [hunk 3]
```diff
@@ -207,7 +207,7 @@
     "description": "The label for the Search Web button and menu items on iOS."
   },
 
-  "shareButtonLabel": "Share...",
+  "shareButtonLabel": "Share",
```
Comment (HansMuller) [inline packages/flutter_localizations/lib/src/l10n/material_en.arb:210]: I may be misunderstanding this but according to this PR's description, `This PR updates the Material label for the share option, currently it is "Share..." (the expected label on iOS), on Android it should be "Share".` So it seems like this value shouldn't change?

**Comment by HansMuller** [comment]
> This may need to go through google internal translation pipeline. @HansMuller Do you know who is taking care of that now? or is there a doc on the process?

Docs are in the comments at the top of files listed below:

If we end up _adding_ messages to material_en.arb then this tool needs to be run to ensure that to-be-translated placeholders exist for all locales:
dev/tools/localization/bin/gen_missing_localizations.dart

If translations are updated in between releases then this tool needs to be run. Ordinarily, it's run by someone on the release team:
dev/tools/localization/bin/gen_localizations.dart

File: packages/flutter_localizations/test/material/translations_test.dart (hunk 2 above)
Comment (bleroux) [reply packages/flutter_localizations/test/material/translations_test.dart:538]: Because this was a test I wrote some months ago (https://github.com/flutter/flutter/pull/130333/files) and where I made a cut and paste mistake (in test/material/translations_test.dart, we should rely on `GlobalMaterialLocalizations`, in that PR I updated both the cupertino test and the material test, and I wrongly copy/pasted this line.

File: packages/flutter_localizations/lib/src/l10n/material_en.arb (hunk 3 above)
Comment (bleroux) [reply packages/flutter_localizations/lib/src/l10n/material_en.arb:210]: Sorry if this PR description is confusing, this change is the purpose of this PR: displaying 'Share' instead of 'Share...'.
'Share' is the native Android label, 'Share...' is the native iOS label.

This is similar to the select all label which is 'Select all' in Material localizations (same as native Android) and 'Select All' in Cupertino localizations (same as native iOS) :

Select all label for Material:
https://github.com/flutter/flutter/blob/89edac404184db1dc9a04b4bad615e28d946037b/packages/flutter/lib/src/material/material_localizations.dart#L1189

Select all label for Cupertino:
https://github.com/flutter/flutter/blob/89edac404184db1dc9a04b4bad615e28d946037b/packages/flutter/lib/src/cupertino/localizations.dart#L486

**Comment by bleroux** [comment]
> If translations are updated in between releases then this tool needs to be run. Ordinarily, it's run by someone on the release team: dev/tools/localization/bin/gen_localizations.dart

I followed the step described at https://github.com/flutter/flutter/blob/master/packages/flutter_localizations/README.md#updating-an-existing-string
(I updated `material_en.arb` and ran the `gen_localizations` script).

File: packages/flutter_localizations/lib/src/l10n/material_en.arb (hunk 1 above)
Comment (bleroux) [reply packages/flutter_localizations/lib/src/l10n/material_en.arb]: I upated the PR to remove the inaccurate 'on iOS' reference in those button label descriptions.

File: packages/flutter_localizations/lib/src/l10n/material_en.arb (hunk 3 above)
Comment (HansMuller) [reply packages/flutter_localizations/lib/src/l10n/material_en.arb:210]: Thanks for the explanation. It was the `description` of the shareButtonLabel message that threw me off; looks like you've already fixed that.

**Comment by bleroux** [comment]
@HansMuller Do you find out who can proceed with the approval of this change and push this through google internal translation pipeline?

**Review by chunhtai — APPROVED** [review]
LGTM
